internal/docgen/parsing_test.go

```diff
@@ -0,0 +1,94 @@
+package docgen
+
+import "testing"
+
+func TestPrefixRuleIDWithPath(t *testing.T) {
```

Comment:
suggestion (non-blocking): should we change these variables' names? (tt, s, name, want)